Assets/Scripts/FloatScreen.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class FloatScreen : MonoBehaviour
+{
+    [SerializeField] private GameObject ToBeFloat = null;
+    [SerializeField] private GameStateManager.GameStates ChangingState;
+    void Start()
+    {
+        GameStateManager.Instance.RegisterOnStateChange(ChangingState, load_scene);
+    }
+
+    private void load_scene(){
+        ToBeFloat.SetActive(true);
+    }
+}
```

Comment:
It's good practice to include a newline at the end of code files. If we ever need to concatenate file, or do a diff or other analysis tools on the code, they usually expect a newline at the end.